contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -6391,6 +6584,691 @@ bbf_ExecDropStmt(DropStmt *stmt)
 	}
 }
 
+/*
+ * bbf_create_partition_tables
+ *	This function creates partitions of babelfish partitioned table
+ *	using the partition scheme and partitioning column.
+ */
+static void
+bbf_create_partition_tables(CreateStmt *stmt)
+{
+	Relation	rel;
+	HeapTuple	tuple;
+	SysScanDesc	scan;
+	ScanKeyData	scanKey[2];
+	char		*input_parameter_type;
+	char		*partition_function_name;
+	Datum		*range_values;
+	Datum		*datum_values;
+	bool		*nulls;
+	int		nelems;
+	Oid		sql_variant_type_oid;
+	Oid		input_type_oid;
+	ListCell	*elements;
+	Oid		partition_column_typoid = InvalidOid;
+	Oid		partition_column_basetypoid = InvalidOid;
+	char		*partition_column_typname = NULL;
+	bool		is_binary_datatype = false;
+	int16		dbid = get_cur_db_id();
+	char		*partition_scheme_name = stmt->partspec->tsql_partition_scheme;
+	char		*relname = stmt->relation->relname;
+	char		*partition_colname = linitial_node(PartitionElem, stmt->partspec->partParams)->name;
+	CreateStmt	*partition_stmt;
+	PlannedStmt	*wrapper;
+	char		*unique_hash;
+	char		*physical_schema_name;
+	char		*logical_schema_name;
+	ArrayType	*values;
+	bool		isnull;
+	int		i;
+	LOCAL_FCINFO(fcinfo, 1);
+
+	/*
+	 * Get partition function name for the provided partition scheme,
+	 * if provided partition scheme exists in current database.
+	 */
+	partition_function_name = get_partition_function_name(dbid, partition_scheme_name);
+	
+	/* Raise error if provided partition scheme doesn't exists in current database. */
+	if (!partition_function_name)
+	{
+		ereport(ERROR, 
+			(errcode(ERRCODE_UNDEFINED_OBJECT), 
+				errmsg("Invalid partition scheme '%s' specifed.", partition_scheme_name)));
+	}
+
+	/* Extract the datatype of partitioning column from CREATE statement. */
+	foreach (elements, stmt->tableElts)
+	{
+		Node		*element = lfirst(elements);
+		ColumnDef	*coldef;
+
+		if (nodeTag(element) != T_ColumnDef)
+			continue;
+
+		coldef = castNode(ColumnDef, element);
+
+		if (pg_strcasecmp(coldef->colname, partition_colname) == 0)
+		{
+			Type ctype = LookupTypeName(NULL, coldef->typeName, NULL, true);
+			Form_pg_type pg_type = (Form_pg_type) GETSTRUCT(ctype);
+
+			partition_column_typoid = pg_type->oid;
+			partition_column_basetypoid = pg_type->typbasetype;
+			partition_column_typname = pstrdup(NameStr(pg_type->typname));
+			ReleaseSysCache(ctype);
+			break;
+		}
+	}
+
+	/* Get OID of sql_variant type. */
+	sql_variant_type_oid = (*common_utility_plugin_ptr->get_tsql_datatype_oid) ("sql_variant");
```

Comment:
We dont have a utility directly for this?

Reply:
no, we don't have any such utility.